### PR TITLE
Proposal: Community Team Structure

### PR DESCRIPTION
**Pull request by mikeal**
I'd like to propose a re-working of the team structure around community stuff.

First and foremost, I'm collapsing into a single group for now. As that group grows we can break out along the lines noted but for the time being it's easier to organize it as a single WG.

The idea here is that all of the activities we're partaking in where the primary goal is to grow a community come under one umbrella. For now, I removed the maintenance of the websites as it seemed like a bit much until we have a real editorial team with contributors. 

It's also worth noting that I think documentation, to some extent, needs to also live with project developers. The way things were worded before gave me the impression that project engineers weren't expected to document APIs they build or change and that it falls on this other group. That isn't very practical, initial docs for features should ship **with the feature**, with the long term improvement of those docs being a community endeavor.

The good news is that this structure gives it all a captain, me :)

### CODE REVIEW

**Comment by meiqimichelle** [comment]
@mikeal I'm not sure you meant it this way, given the what you wrote in your description -- your PR breaks this work into the following **four** working groups under a general category of `Community`. Was that your intention?

    - Evangelism (Events, Social Media)
    - Editorial (Blogs, Documentation)
    - Education (ProtoSchool, Demos)
    - Automation (Probots, etc)

In general, though, I want to give this a big 👍 -- thank you, @mikeal ! I like the collapse/clarification of this work into one word: `Community`. I also really appreciate the reminder that documentation, to some extent, should be a part of what project developers do. (I don't think the old WG was meant to give the impression that they *don't* do this, but I like the explicit-ness of removing "documentation" as a separate thing.)

**One question:** in this structure, would more people with Rob's skillset be a part of what you call the Editorial team? Or are you saying that they should just be members of specific working groups along with everyone else? Or maybe a bit of column A, column B :)

**One comment:** I am worried that this structure is still missing a vital workstream, which is the hard work and specialized skillset it takes to look across all our words and illustrations and websites and other properties to create an information architecture and brand that makes sense. The result of that work: people will be able find and understand information (relatively) accurately and quickly -- so that they can *then* contribute.

This systems-level clarification work will not be done by outside contributors either at the start or in the long run -- it's an activity that needs care, feeding, and leadership over time to create and invite collaboration on styleguides, information architectures, guides, etc. **We should not wait until we have an editorial team to start doing that work**. (The website tasks, as I see them, are not about maintenance at this point. They are more about this type of hard work.)

Do you see this type of role as a part of `Community`? (I could also see this type of work being done at the PL level so it would have insight into PL branding/information architecture, but it might be easier to start with IPFS and go from there.) If it is no longer here, then we should create a new working group to address this need.

**Comment by mikeal** [comment]
> I'm not sure you meant it this way, given the what you wrote in your description -- your PR breaks this work into the following four working groups under a general category of Community.

I was worried about this. It was **not** meant to be interpreted this way :)

I was trying to find a way of collapsing the work into a single group but calling out all the items that are in the WG. That has apparently failed.

> in this structure, would more people with Rob's skillset be a part of what you call the Editorial team? Or are you saying that they should just be members of specific working groups along with everyone else? Or maybe a bit of column A, column B :)

While I've worked with Rob on a few things I'm not familiar with scope and OKR's assigned to that work so I can't definitively categorize it.

I've been trying to find the best way to dilettante the lines between "Project/Product" and "Community" resourcing and there isn't always a clear line. I think the best way to split it up right now is to say that there is a product roadmap and a community roadmap. The tasks we (Community Ops) can take on directly in the short and mid-term are limited and we'll end up prioritizing the things that we think are low hanging fruit in terms of growing a community. That won't always align with the work that is being prioritized on the product side. For instance, PL is prioritizing cluster for a lot of very good reasons but cluster wouldn't be a short term high priority target for community building yet. We know that there are much larger impact things we can do elsewhere that we need to prioritize before we work on stuff like cluster and if we try to do everything at once we'll end up being too spread out to make a real impact.

With that in mind, it's going to be the case that certain documentation work may be prioritized by the product teams ahead of what Community has bandwidth for. When that happens, it should probably live with the PM or the Engineers. The goal isn't, in the short term, to build a community around that documentation it is simply to make sure it is quality and completed by a particular time. Of course the Community side will continue to collaborate and recommend certain tooling choices that keep the door open to a community in the future, but we wouldn't drive it.

The vision for the Editorial team is that *eventually* all of this could live there, but we won't be there for a quite a while. We need to start small and grow from there. The main goal of the Editorial team is, as you may have guessed, to build a community around people with skills in the written word. I'd like to model it after [Hoodie's Editorial Team](https://github.com/hoodiehq/editorial). In the short term we'll build it around specific activities but as the capacity grows I'd love to see it expand across documentation in all our projects, but that is a ways out.

> I am worried that this structure is still missing a vital workstream, which is the hard work and specialized skillset it takes to look across all our words and illustrations and websites and other properties to create an information architecture and brand that makes sense. The result of that work: people will be able find and understand information (relatively) accurately and quickly -- so that they can then contribute.

I have the same concerns but I struggle with the order of operation for trying to fix this. Being that I'm relatively new to PL I don't feel like I'm yet in a position to drive this kind of uniform messaging across the org. All I can do at this point is bite off parts of this problem and hope they contribute in some way to a larger solution.

For instance, the Blog we're building will **not** be branded to a specific project but will be general enough to cover all of our current scope with an identity focused only on decentralization. 

In ProtoSchool we're tackling education around some of the core concepts and identifying areas that are just not understandable by most people. As that happens I'm trying to clean up the messaging as well as building the educational materials to teach it. This should at least contribute to the understand-ability of the whole by making some of the parts easier to understand.

> This systems-level clarification work will not be done by outside contributors either at the start or in the long run -- it's an activity that needs care, feeding, and leadership

I agree, but one thing we struggle with as we get our heads deeper into the weeds of decentralization is drawing in the perspective people who aren't this deep. As we've been hiring in Community Ops we've specifically looked for people that have a track record of maintaining that kind of perspective. While I don't think we'll be **driving** this systems-level clarification I do think that we'll end up being able to bring valuable contributions to its development.

> We should not wait until we have an editorial team to start doing that work.

+1

> The website tasks, as I see them, are not about maintenance at this point. They are more about this type of hard work.

Ya, websites were the one thing I just outright removed from the description right now. Websites will always include cross-collaboration between many groups but if anyone was going to "own" them I'd recommend program management.

**Comment by meiqimichelle** [comment]
> With that in mind, it's going to be the case that certain documentation work may be prioritized by the product teams ahead of what Community has bandwidth for. When that happens, it should probably live with the PM or the Engineers. 

The Engineers and PMs don't have bandwidth for all this, either, and in any case don't always have the skillset or time to tackle your next point....

> but one thing we struggle with as we get our heads deeper into the weeds of decentralization is drawing in the perspective people who aren't this deep. 

+100 to that! This is why I'm trying to hire people who have the specific skillset to talk to product end users and clarify/write for and with them (by testing content with end users, learning what doesn't make sense, and iterating our illustrations/explanations/information architectures). I think what you're saying is that you're hiring a similar skillset, but that those people wouldn't be available, say, to help the JavaScript Core Dev WG create clear documentation in plain language, for example, or guides on their core concepts, because they'll be busy with their community-building work (though your Community work should, in the end, contribute to the greater mission of clarification and helping people understand our concepts, etc).

> Being that I'm relatively new to PL I don't feel like I'm yet in a position to drive this kind of uniform messaging across the org. All I can do at this point is bite off parts of this problem and hope they contribute in some way to a larger solution. ...I don't think we'll be driving this systems-level clarification I do think that we'll end up being able to bring valuable contributions to its development.

Got it. It sounds like, then, this is still an open problem that another WG or team (not Community) should tackle. (One side note: solving this problem doesn't need to be related to how long someone has been in the org. Brand or design strategists are called on all the time to facilitate conversations and then design visual/communication solutions for this sort of problem).

> if anyone was going to "own" them I'd recommend program management.

This clarification work seems like a hot potato that no one is addressing, unfortunately.

After this chat (thank you @mikeal !) I think I understand where you're heading with this PR (and I'll wait on your clarification on the layout to make it one WG instead of four :) ). And! Separately! I think we also need to find a home for the other work that you've made clear that would *not* be in Community scope, to include:

- User-centered content folks working on product WG priorities
- A higher-level WG or team to tackle the systems-level clarification stuff -- where this would be is probably related to my confusion about where [cross-PL work should live](https://github.com/protocol/LabOS/issues/13).

File: TEAM_STRUCTURES.md [hunk 1]
```diff
@@ -12,9 +12,11 @@
...
     - [IPFS Infrastructure](#ipfs-infrastructure)
     - [Dev Team Enablement (QA, Testing, Support)](#qa-testing-and-dev-team-enablement)
-  - **Community, Education and Usage:**
-    - [Community, Events & Evangelism](#community-events-evangelism)
-    - [Documentation, Education and Websites](#documentation-education-and-websites)
+  - **[Community](#community):**
+    - Evangelism (Events, Social Media)
+    - Editorial (Blogs, Documentation)
+    - Education (ProtoSchool, Demos)
+    - Automation (Probots, etc)
```
Comment (diasdavid) [inline TEAM_STRUCTURES.md]: If these are not working Groups and just areas of focus of the Community Working Group, let's delete these bullets from the table of contents to not mislead into thinking that there are 4 WGs

File: TEAM_STRUCTURES.md [hunk 2]
```diff
@@ -60,9 +62,11 @@ A byproduct of both of these team structures achieves another important goal: ma
...
   - [IPFS Infrastructure](#ipfs-infrastructure)
   - [Dev Team Enablement (QA, Testing, Support)](#qa-testing-and-dev-team-enablement)
-- **Community, Education and Usage:**
-  - [Community, Events & Evangelism](#community-events-evangelism)
-  - [Documentation, Education and Websites](#documentation-education-and-websites)
+- **[Community](#community):**
+  - Evangelism (Events, Social Media)
+  - Editorial (Blogs, Documentation)
+  - Education (ProtoSchool, Demos)
+  - Automation (Probots, etc)
```
Comment (diasdavid) [inline TEAM_STRUCTURES.md]: Same as above

File: TEAM_STRUCTURES.md [hunk 3]
```diff
@@ -131,38 +135,26 @@ The QA, Testing and Dev Team Enablement Working Group focuses on building develo
 - Define process and help achieve high quality sustainable open source modules.
 - Automating tasks for things such as: release cycle, ensuring code quality, activity reports.
 
-### Community, Events & Evangelism
+### Community
 
 > This Working Group hasn't been formed yet.
```
Comment (diasdavid) [inline TEAM_STRUCTURES.md]: Would you say that with this PR the Working Group is formed?

File: TEAM_STRUCTURES.md [hunk 4]
```diff
@@ -131,38 +135,26 @@ The QA, Testing and Dev Team Enablement Working Group focuses on building develo
...
 
 - **Coordination**: https://github.com/ipfs/community
 - **Captain**: [Mikeal Rogers](https://github.com/mikeal)
 
-The Community, Events & Evangelism Working Group is responsible for building an environment that fosters collaboration, innovation, and inclusiveness for the IPFS community.
+Community outreach working group. Coordinates the communities around events, blog posts
+documentation, automation and education.
```
Comment (diasdavid) [inline TEAM_STRUCTURES.md]: Move the 4 verticals of focus of this working group to this section.

**Comment by mikeal** [comment]
> It will be refreshing to restart both work streams under one umbrella that adopts the good habits that the other Working Groups have developed

Speaking of, are these habits/practices documented somewhere? This doc leaves them fairly open and I've certainly seen many more best practices than it references like regularly recorded zoom calls.

File: TEAM_STRUCTURES.md (hunk 3 above)
Comment (mikeal) [reply TEAM_STRUCTURES.md]: Yes :)

**Comment by diasdavid** [comment]
> are these habits/practices documented somewhere?

I did a presentation on them a month ago for the team and I plan on recording a new version to share with the whole group. I'll share it with you meanwhile.

Nevertheless, one thing that is important to note is that we are deliberately under-prescriptive when it comes to the comms and structure that each Working Group uses. The only requirement is that each WG projects their progress using OKRs (the common interface to eval progress in the org) and it carries the values of the entire org, specially with regards to openness and async communication to enable others to take part in it too.

**Comment by mikeal** [comment]
> Nevertheless, one thing that is important to note is that we are deliberately under-prescriptive

Yup, I appreciate this, I'd just like to document things we've found that we've found work well so that WG's can more easily adopt practices they think will be helpful. After you record the presentation send me a link :)

**Comment by steverichmond** [comment]
David, if you would also also share the link with me I would be grateful.
This is a topic in LabOS - there are some different WG practices forming in
different teams, and your take on this would be very helpful.

**Comment by meiqimichelle** [comment]
I have thought about this more and have a slight adjustment. 

I agree that @mikeal should collapse the overlap in how those two working groups were originally defined, and take over the work that is part of his vision for CommunityOps. I am very happy about this! However, through conversation in this chain, we identified two initiatives that were a part of the old Documentation WG that Mikeal explicitly said will *not* be a part of his work. To @diasdavid 's point about creating WGs when needed, these were *already* identified as 'mitosis' from other WGs, namely:

- Documentation folks working on product WG priorities (rather than Community priorities) (aka Rob's skillset)
- A higher-level WG or team to coordinate the systems-level clarification stuff (cross-PL websites, branding, messaging, etc)

So, I don't agree that the second working group should completely disappear -- that other work needs to happen somewhere. **I propose moving the website work to a newly-minted Design Team as per [this LabOS issue](https://github.com/protocol/LabOS/issues/13#issuecomment-420449235), and the product-focused documentation work to stay in a separate Documentation WG.**

**Comment by meiqimichelle** [comment]
Update: @mikeal and I are going to align timezones and make sure we're on the same page. So, comment if you like -- but I'm also requesting a day or so for me to catch up with Mikeal, might have an update for this chain after that. Watch this space (tm).

**Comment by meiqimichelle** [comment]
Ok! Hi all. MHz here. Just got off a very helpful call with @mikeal . This is where we're at:

- The end goal of what both of us want to see is the same: helping new IPFS community members understand our project so they can contribute. We didn't realize that before. So, @mikeal 's team *will* be covering the tasks/items that I was afraid were going to get dropped. 👍 to combining the WGs.

- There is *still* a separate need around building out capacity on the Design Team. This might include helping to sort out cross-PL websites, branding, messaging -- eventually. We both recognize that right *now* might not be the time to move quickly on that, as we're still building out the Communication WG at the LabOS level. But there's still a current need to build the team so that @mikeal can be effective in his Community work (and not to mention all the other work we do across PL). He suggested that in particular we need more interaction design support.

In summary, I'm 👍 on combining the WGs under Mikeal's Community WG, and will move the Design Team discussion to [this LabOS](https://github.com/protocol/LabOS/issues/13#issuecomment-421461555) issue.
